starter-code/contact-app/src/app/contact-list/contact-list.component.html

```diff
@@ -1,7 +1,64 @@
-<div>
-  <label for="contact-name"></label>
-  <input type="text" [(ngModel)]="newContact.name">
+<ul *ngFor='let contact of contacts'>
+  <li>Name: {{contact.name}}</li>
+  <li>Email: {{contact.email}}</li>
+  <li>Phone number: {{contact.phoneNumber}}</li>
+  <!-- <img [src]='contact.image'> -->
+</ul>
+
+<form action=""></form>
+
+<form 
+  (ngSubmit)="submitForm(form)" 
+  #form="ngForm"
+  novalidate>
+
+  <div class="field"
+    [ngClass]="{'has-error': showErrorMessages && contactName.errors}">
+    <label for="contact-name">Name:</label>
+    <input type="text" 
+    name = "contact-name"
+    [(ngModel)]="newContact.name"
+    #contactName="ngModel"
+    required
+    [disabled]="processing">
+    <div class="error-messages"
+      *ngIf="showErrorMessages && contactName.errors">
+      <p *ngIf="contactName.errors.required"
+        class="validation">
+        Contact name required
+      </p>
+    </div>
+  </div>
+
+  <div class="field"
+  [ngClass]="{'has-error': showErrorMessages && contactEmail.errors}">
+  <label for="contact-email">Email:</label>
+  <input type="email" 
+  name = "contact-email"
+  [(ngModel)]="newContact.email"
+  #contactName="ngModel"
+  required
+  [disabled]="processing">
+  <div class="error-messages"
+    *ngIf="showErrorMessages && contactEmail.errors">
+    <p *ngIf="contactEmail.errors.required"
+      class="validation">
+      Email required
+    </p>
+  </div>
 </div>
+
+
+
+  <button type="submit" [disabled]="processing">Submit</button>
```

Comment:
Review the structure that you have in the cheat sheet: https://github.com/ironhack/bcn-webdev-cheatsheet/tree/master/m3/angular-forms